training/src/anemoi/training/train/tasks/base.py

```diff
@@ -593,3 +596,21 @@ def configure_optimizers(self) -> tuple[list[torch.optim.Optimizer], list[dict]]
         )
 
         return [optimizer], [{"scheduler": scheduler, "interval": "step"}]
+
+    def setup(self, stage: str) -> None:
+        """Lightning hook that is called after model is initialized but before training starts."""
+        # The two ifs should be separate, but are combined due to pre-commit hook
+        if stage == "fit" and self.trainer.is_global_zero:
+            # Log variable scaling on rank 0
+            max_variables_to_log = (
+                self.config.diagnostics.log.mlflow.max_num_variable_scalings
+                if hasattr(self.config.diagnostics.log.mlflow, "max_num_variable_scalings")
+                else 200
```

Comment:
this 200 could be a constant defined for mlflow under the training/diagnostics/mlflow/logger.py as that's just relevant for this